File: src/Sketch.java
import processing.core.PApplet;
import gab.opencv.*;
import processing.video.*;
import java.awt.*;


public class Sketch extends PApplet {

    Capture video;
    OpenCV opencv;

    @Override
    public void setup() {
        size(640, 480);
        video = new Capture(this, 640/2, 480/2);
        opencv = new OpenCV(this, 640/2, 480/2);
        opencv.loadCascade(OpenCV.CASCADE_FRONTALFACE);

        video.start();
    }

    @Override
    public void draw() {
        scale(2);
        opencv.loadImage(video);

        image(video, 0, 0 );

        noFill();
        stroke(0, 255, 0);
        strokeWeight(3);
        Rectangle[] faces = opencv.detect();
        println(faces.length);

        for (Rectangle face : faces) {
            println(face.x + "," + face.y);
            rect(face.x, face.y, face.width, face.height);
        }
    }

    public void captureEvent(Capture c) {
        c.read();
    }

    @Override
    public boolean sketchFullScreen() {
        return true;
    }
}

File: src/main/scala/Sketch.scala
import processing.core.PApplet
import gab.opencv._
import processing.video.Capture
import java.awt.Rectangle


class Sketch extends PApplet {

  val video = new Capture(this, 640 / 2, 480 / 2)
  val opencv = new OpenCV(this, 640 / 2, 480 / 2)

  override def setup() {
    size(640, 480)
    opencv.loadCascade(OpenCV.CASCADE_FRONTALFACE)
    video.start()
  }

  override def draw() {
    scale(2)
    opencv.loadImage(video)
    image(video, 0, 0)
    noFill()
    stroke(0, 255, 0)
    strokeWeight(3)
    val faces: Array[Rectangle] = opencv.detect
//    println(faces.length)
    for (face <- faces) {
//      println(face.x + "," + face.y)
      rect(face.x, face.y, face.width, face.height)
    }
  }

  def captureEvent(c: Capture) = c.read()

  override def sketchFullScreen: Boolean = true
}
